chia/wallet/wallet_node.py

```diff
@@ -1041,11 +1038,7 @@ async def new_peak_from_trusted(self, new_peak_hb: HeaderBlock, latest_timestamp
             if peer.peer_node_id not in self.synced_peers:
                 if new_peak_hb.height - current_height > self.LONG_SYNC_THRESHOLD:
                     self.wallet_state_manager.set_sync_mode(True)
-                self._primary_peer_sync_task = asyncio.create_task(
-                    self.long_sync(new_peak_hb.height, peer, uint32(max(0, current_height - 256)), rollback=True)
-                )
-                await self._primary_peer_sync_task
-                self._primary_peer_sync_task = None
+                await self.long_sync(new_peak_hb.height, peer, uint32(max(0, current_height - 256)), rollback=True)
```

Comment:
@xdustinface i mean now if you shut down when we are waiting for long sync i dont' think there is any place where we break, it looks like we would wait for the sync to complete

Reply:
We still cancel the task which is awaiting the `long_sync`, the `_process_new_subscriptions_task` and once this is canceled there will be a canceled exception raised just the same like if we canceled the `_primary_peer_sync_task` before, right?

Reply:
if thats the case then thats fine, it looks like someone went into some effort to make ut possible we can explicitly cancel that part i dident debug if you say this shuts down ok during long sync that thats good with me